rasa_nlu_examples/classifiers/sparse_sklearn_intent_classifier.py

```diff
@@ -0,0 +1,267 @@
+        )
+
+    def _num_cv_splits(self, y: np.ndarray) -> int:
+        folds = self.component_config["max_cross_validation_folds"]
+        return max(2, min(folds, np.min(np.bincount(y)) // 5))
+
+    def _create_classifier(
+        self, num_threads: int, y: np.ndarray
+    ) -> "sklearn.model_selection.GridSearchCV":
+        from sklearn.model_selection import GridSearchCV
+        from sklearn.naive_bayes import BernoulliNB
+
+        alpha = self.component_config["alpha"]
+        binarize = self.component_config["binarize"]
+        fit_prior = self.component_config["fit_prior"]
+        # dirty str fix because sklearn is expecting
+        # str not instance of basestr...
+        tuned_parameters = [
+            {"alpha": alpha, "binarize": binarize, "fit_prior": fit_prior}
+        ]
+
+        # aim for 5 examples in each fold
+
+        cv_splits = self._num_cv_splits(y)
+
+        return GridSearchCV(
+            BernoulliNB(alpha=1, binarize=0.0, fit_prior=True),
+            param_grid=tuned_parameters,
+            n_jobs=num_threads,
+            cv=cv_splits,
+            scoring=self.component_config["scoring_function"],
+            verbose=1,
+            iid=False,
+        )
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        """Return the most likely intent and its probability for a message."""
+
+        if not self.clf:
+            # component is either not trained or didn't
+            # receive enough training data
+            intent = None
+            intent_ranking = []
+        else:
+            X = self._get_sentence_features(message)
+            intent_ids, probabilities = self.predict(X)
+            intents = self.transform_labels_num2str(np.ravel(intent_ids))
+            # `predict` returns a matrix as it is supposed
+            # to work for multiple examples as well, hence we need to flatten
+            probabilities = probabilities.flatten()
+            if intents.size > 0 and probabilities.size > 0:
+                ranking = list(zip(list(intents), list(probabilities)))[
+                    :LABEL_RANKING_LENGTH
+                ]
+
+                intent = {"name": intents[0], "confidence": probabilities[0]}
+
+                intent_ranking = [
+                    {"name": intent_name, "confidence": score}
+                    for intent_name, score in ranking
+                ]
+            else:
+                intent = {"name": None, "confidence": 0.0}
+                intent_ranking = []
+
+        message.set("intent", intent, add_to_output=True)
+        message.set("intent_ranking", intent_ranking, add_to_output=True)
+
+    def predict_prob(self, X: scipy.sparse.spmatrix) -> np.ndarray:
+        """Given a bow vector of an input text, predict the intent label.
+        Return probabilities for all labels.
+        :param X: bow of input text
+        :return: vector of probabilities containing one entry for each label"""
```

Comment:
Nitpick. But doens't the styleguide prefer

```
"""
Given a bow vector of an input text, predict the intent label.
        Return probabilities for all labels.
        :param X: bow of input text
        :return: vector of probabilities containing one entry for each label
"""
```

instead of 

```
"""Given a bow vector of an input text, predict the intent label.
        Return probabilities for all labels.
        :param X: bow of input text
        :return: vector of probabilities containing one entry for each label"""
```